packit_service/worker/build/build_helper.py

```diff
@@ -373,9 +417,30 @@ def _report(
         The status reporting should be done through this method
         so we can extend it in subclasses easily.
         """
-        self.status_reporter.report(
-            description=description, state=state, url=url, check_names=check_names,
-        )
+        if isinstance(self.project, GitlabProject) and not self.is_reporting_allowed:
+            # "Packit-User" does not have access to project.
+            self.request_project_access()
+            self.report_status_by_comment(state, url, check_names)
+        else:
+            self.status_reporter.report(
+                description=description, state=state, url=url, check_names=check_names,
+            )
+
+    def report_status_by_comment(
+        self, state: CommitStatus, url: str, check_names: Union[str, list, None],
+    ) -> None:
+        """
+        Reporting build status with MR comment if no permission to the fork project
+        """
+        build_states = [CommitStatus.error, CommitStatus.success, CommitStatus.failure]
+
+        if isinstance(check_names, str):
+            check_names = [check_names]
+
+        for check in check_names:
+            comment_msg = f"[{check}]({url}): {state.name.upper()}"
+            if state in build_states:
+                self.project.pr_comment(self.pr_id, comment_msg)
```

Comment:
Can we merge those to one comment to not be so spammy?

Reply:
Done!